### PR TITLE
Iterate on shortfin/README.md.

### PR DESCRIPTION
**Pull request by ScottTodd**
Progress on https://github.com/nod-ai/SHARK-Platform/issues/359.

This page will appear at https://pypi.org/project/shortfin/, so include enough context for users. I'm still deciding where it makes sense to highlight current support status for platforms, operating systems, models, etc.

### CODE REVIEW

File: shortfin/README.md [hunk 1]
```diff
@@ -1,20 +1,44 @@
...
+* The "libshortfin" inference library written in C/C++ and built on
+  [IREE](https://github.com/iree-org/iree)
+* Python bindings for the underlying inference library
+* Example applications in
+  ['shortfin_apps'](https://github.com/nod-ai/SHARK-Platform/tree/main/shortfin/python/shortfin_apps)
+  built using the python bindings
 
+## Prerequisites
+
+* Python 3.11+
```
Comment (marbre) [inline shortfin/README.md:15]: Unit tests pass with 3.10 and we build wheels for 3.10. However, we should probably test the apps for which we say that those are supported.

File: shortfin/README.md [hunk 2]
```diff
@@ -1,20 +1,44 @@
...
 ```
+
+## Developer guides
+
+### Install local packages
+
+After cloning this repository, from the `shortfin/` directory:
+
+```bash
 python -m pip install .
```
Comment (marbre) [inline shortfin/README.md:32]:
```suggestion
python -m pip install -e .
```
Editable?

File: shortfin/README.md [hunk 3]
```diff
@@ -167,16 +191,16 @@ recommended:
...
 
 Support for free-threaded Python builds (aka. "nogil") is in progress. It
 is currently being tested via dev builds of CPython 3.13 with the
 `--disable-gil` option set. There are multiple ways to acquire such an
 environment. If using `pyenv`, here is a way:
 
-```
+```bash
 # Build a free-threaded 3.13 version.
 pyenv install --debug 3.13t-dev
```
Comment (marbre) [inline shortfin/README.md:205]:
```suggestion
pyenv install 3.13t
```
We probably want to update this and go over the entire section as Python 3.13t was released in the meantime and there is no (strict) need to install a dev build.

Comment (ScottTodd) [reply shortfin/README.md]: Good catch. Updated instructions for pyenv.

File: shortfin/README.md (hunk 2 above)
Comment (ScottTodd) [reply shortfin/README.md]: Sure. Reworked into a "quick start" step and added `python -m pip install -r requirements-tests.txt`.

File: shortfin/README.md (hunk 1 above)
Comment (ScottTodd) [reply shortfin/README.md:15]: I heard someone say 3.11+ this morning, so starting there. We can expand when we have better test / QA coverage.

File: shortfin/README.md [hunk 4]
```diff
@@ -1,20 +1,52 @@
...
+## Prerequisites
+
+* Python 3.11+
+
+## Simple user installation
+
+Install the latest stable version:
+
+```bash
+python -m pip install shortfin
```
Comment (marbre) [inline shortfin/README.md:22]:
```suggestion
pip install shortfin
```
We may want to harmonize this with the root README.

File: shortfin/README.md [hunk 5]
```diff
@@ -1,20 +1,52 @@
...
-python -m pip install .
+
+## Developer guides
+
+### Quick start: install local packages and run tests
+
+After cloning this repository, from the `shortfin/` directory:
+
+```bash
+python -m pip install -e .
```
Comment (marbre) [inline shortfin/README.md:32]:
```suggestion
pip install -e .
```

File: shortfin/README.md [hunk 6]
```diff
@@ -1,20 +1,52 @@
...
 ```
 
-Run tests:
+Install test requirements:
 
+```bash
+python -m pip install -r requirements-tests.txt
```
Comment (marbre) [inline shortfin/README.md:38]:
```suggestion
pip install -r requirements-tests.txt
```

File: shortfin/README.md [hunk 7]
```diff
@@ -1,20 +1,52 @@
...
 ```
+
+Run tests:
+
+```bash
 python -m pytest -s tests/
```
Comment (marbre) [inline shortfin/README.md:44]:
```suggestion
pytest -s tests/
```

File: shortfin/README.md [hunk 8]
```diff
@@ -167,23 +193,23 @@ recommended:
...
-python -c 'import sys; print(sys._is_gil_enabled())'
-```
+    ```bash
+    # Install a free-threaded 3.13 version.
+    pyenv install 3.13t
 
-Further ways of installing a free-threaded CPython interpreter are documented at
-[py-free-threading.github.io](https://py-free-threading.github.io/installing_cpython/).
+    # Test (should print "False").
+    pyenv shell 3.13
```
Comment (marbre) [inline shortfin/README.md:213]:
```suggestion
    pyenv shell 3.13t
```
At least if 3.13 and 3.13t are installed, you need the `t`.

**Review by marbre — CHANGES_REQUESTED** [review]

**Comment by marbre** [comment]
Thanks for further iterating on this Scott. Looks pretty good.

File: shortfin/README.md (hunk 4 above)
Comment (ScottTodd) [reply shortfin/README.md:22]: Sure. In some cases I like to be explicit with `python -m`, but user docs are likely worth trimming to the shorter form.

**Review by marbre — APPROVED** [review]

**Pull request merged**